packages/mongodb-rag-core/src/conversations/MongoDbConversations.ts

```diff
@@ -164,9 +164,9 @@ export function makeMongoDbConversationsService(
 
 export function createMessage(messageParams: AddSomeMessageParams) {
   const message = {
-    id: new ObjectId(),
-    createdAt: new Date(),
     ...messageParams,
+    createdAt: new Date(),
+    id: messageParams.id ?? new ObjectId(),
```

Comment:
Why this switch? What if we want to override the current date for `createdAt`, e.g. in a test (not actually sure if/when we do this)

Reply:
it was for capturing the messageParams.id, which is needed for the tracing